From 95d7c29bdff4847fb039de195f3896264bfad91c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 10 Mar 2016 14:05:37 +0000
Subject: [PATCH] fix for 10 sec hang with audio-only formats with gstreamer1

just do the same as we did for gstreamer 0.10 and set the
size found condition once we get the message even if there is no size

Change-Id: I826d76dd1097432e5bd33d4acc1a187b8e5068da
(cherry picked from commit 702790258c6b3ec05ce04f7c2839a48776381535)
---
 avmedia/source/gstreamer/gstplayer.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/avmedia/source/gstreamer/gstplayer.cxx b/avmedia/source/gstreamer/gstplayer.cxx
index d12e8e9..b820ae1 100644
--- a/avmedia/source/gstreamer/gstplayer.cxx
+++ b/avmedia/source/gstreamer/gstplayer.cxx
@@ -549,11 +549,12 @@ GstBusSyncReply Player::processSyncMessage( GstMessage *message )
 
                     DBG( "queried size: %d x %d", mnWidth, mnHeight );
 
-                    maSizeCondition.set();
                 }
                 gst_caps_unref( caps );
                 g_object_unref( pad );
             }
+
+            maSizeCondition.set();
         }
 #endif
     } else if (gst_is_missing_plugin_message(message)) {
-- 
2.7.1

